src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -141,6 +141,12 @@ public TracerSettings(IConfigurationSource source)
             HeaderTags = HeaderTags.Where(kvp => !string.IsNullOrEmpty(kvp.Key) && !string.IsNullOrEmpty(kvp.Value))
                                    .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
 
+            var serviceNameMappings = source?.GetDictionary(ConfigurationKeys.ServiceNameMappings)
+                                      ?.Where(kvp => !string.IsNullOrEmpty(kvp.Key) && !string.IsNullOrEmpty(kvp.Value))
```

Comment:
It should be `IsNullOrWhitespace` I believe, since we're going to trim the value afterwards
(and I just realized the other configuration keys have the same issue)

Reply:
Good point, I'll fix that and the other configuration keys too

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1142/commits/1967a3896d3f9f9779bb2b76218b6253b7dbf3b1